doc/plutus/tutorials/basic-validators.rst

```diff
@@ -1,133 +0,0 @@
-.. highlight:: haskell
-.. _basic_validators_tutorial:
-
-Writing basic validator scripts
-===============================
-
-:term:`Validator scripts<validator script>` are the programs that can be used to lock transaction outputs on the chain.
-Validator scripts are :term:`Plutus Core` programs, but we can use :term:`Plutus Tx` to write them easily in Haskell.
-
-Validator arguments
--------------------
-
-Validators receive some information from the validating node:
-
-- The :term:`redeemer`, which is some script-specific data specified by the party spending the output.
-- The :term:`datum`, which is some script-specific data specified by the party who created the output.
-- The :term:`validation context`, which contains a representation of the spending transaction, as well as the index of the input whose validator is currently being run.
-
-The validator is a function which receives these three inputs as *arguments*.
-The validating node is responsible for passing them in and running the validator.
-
-The ``Data`` type
------------------
-
-But how are the validator's arguments passed?
-At least the redeemer and datum can be of different types depending on the script.
-
-The answer is that we pass them as a generic structured data type :hsobj:`PlutusCore.Data.Data`.
-``Data`` is designed to make it easy to encode structured data into it, and to be itself encoded as CBOR.
-
-Consequently, the validator scripts we will write in this tutorial take three arguments of type ``Data``.
-
-However, you will typically not want to use ``Data`` directly in your program, rather you will want to use your own datatypes.
-We can easily convert to and from ``Data`` with the :hsobj:`PlutusTx.IsData.Class.ToData` and :hsobj:`PlutusTx.IsData.Class.FromData` typeclasses.
-
-You usually don't need to write your own ``To/FromData`` instances.
-Instead, you can use the ``makeIsData`` Template Haskell function to generate one.
-
-.. literalinclude:: BasicValidators.hs
```

Comment:
`BasicValidators.hs` should be removed too.

Reply:
Removed

Reply:
Actually after I removed the file `BasicValidators.hs` I got some build errors, so I put it back. However, the file `basic-validators.rst` has been removed.

Reply:
Ah, `BasicValidators.hs` is also used by `doc/plutus/howtos/exporting-a-script.rst`.